### PR TITLE
fix: --smart-group only works for current user

### PR DESCRIPTION
**Pull request by liketurbo**
Relevant issue: #545

### CODE REVIEW

File: src/output/render/groups.rs [hunk 1]
```diff
@@ -53,20 +56,17 @@ impl Render for Option<f::Group> {
...
-                    if current_user.name() == group.name() {
-                        ":".to_string()
-                    } else {
-                        group_name
+        if let GroupFormat::Smart = group_format {
+            if let Some(file_uid) = file_user {
+                if let Some(file_user) = users.get_user_by_uid(file_uid.0) {
+                    let file_user_name = file_user.name().to_string_lossy().to_string();
+                    let group_name_text = group.name().to_string_lossy().to_string();
+                    if file_user_name == group_name_text {
```
Comment (ariasuni) [inline src/output/render/groups.rs:64]: Seems to me like you can just do (at least rust-analyzer is OK with this code):

```suggestion
                    if file_user.name().to_string_lossy() == group.name().to_string_lossy() {
```

**Review by ariasuni — CHANGES_REQUESTED** [review]

File: src/output/render/groups.rs (hunk 1 above)
Comment (liketurbo) [reply src/output/render/groups.rs]: Thanks! It is much better

**Review by MartinFillon — APPROVED** [review]
LGTM

**Review by MartinFillon — APPROVED** [review]

**Review by ariasuni — APPROVED** [review]

**Review by cafkafk — APPROVED** [review]
Integration tests and local testing confirms this as seemingly not introducing any regressions. I can't test this feature directly since my group isn't the same as my user.

**Pull request merged**